### DISCUSSION TITLE
Custom Activity Nullable Attributes

### DISCUSSION THREAD

**Discussion by GaryJurgens** [elsa-workflows/elsa-core#5239] [Q&A]
When writing a Custom Activity, Elsa 3, how can you make an attribute nullable or not required.

Input<string> Myname {Get;Set;}

var Name = MyName.Get(Context), MyName could be empty or null, to invoke different processes in my activity. All what happens now is it throws an Exception stating MyName is required. 

How do I make it so that I can have a null or empty value 

thanks 

G

**Comment by glime-ai** [comment]
To make an attribute nullable or not required in a custom activity in Elsa, you can use the `HasValue` property of the `Input<T>` class to check if the input has been provided before attempting to get its value. This way, you can handle the scenario where the input might be null or empty without throwing an exception.

Here's an example of how you might modify your `Execute` method to handle a nullable `Name` property:

```csharp
protected override void Execute(ActivityExecutionContext context)
{
    // Check if the Name input has a value
    if (Name.HasValue(context))
    {
        var name = Name.Get(context);
        // Continue with the process using the name
    }
    else
    {
        // Handle the case where Name is null or empty
        // You can invoke different processes here
    }
}
```

By using the `HasValue` method, you can conditionally check for the presence of input and avoid exceptions related to accessing a null or empty value. If you want to allow the input to be optional, you should not set a default value for the `Name` property, as this would prevent it from being null.

If you need further assistance or clarification, please let me know.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by GaryJurgens** [reply]
This Does not work in Elsa 3 there is no has value method i can see, unless i am missing something.
